plugins/org.jkiss.dbeaver.core/src/org/jkiss/dbeaver/ui/preferences/PrefPageToolbarCustomization.java

```diff
@@ -53,8 +53,10 @@
 import java.util.stream.Collectors;
 
 
-public class PrefPageToolbarCustomization extends AbstractPrefPage implements IWorkbenchPreferencePage {
-    
+public class PrefPageToolbarCustomization extends TargetPrefPage {
```

Comment:
Why TargetPrefPage? It is designed for connection-specific properties. AbstractPrefPage maybe better.

Reply:
I used AbstractPrefPage before, it doesn't bring this page to property settings